src/App.js

```diff
@@ -0,0 +1,49 @@
+import React, { Component } from 'react';
+import { Route, Switch, withRouter } from 'react-router-dom';
+import { connect } from 'react-redux';
+
+// import logo from './logo.svg';
+import './App.css';
+import './assets/semantic/semantic.min.css';
+// import { Header, Button, Divider } from 'semantic-ui-react';
+
+import * as actions from './store/actions';
+
+import Layout from './components/Layout/Layout';
+import TaskBuilder from './containers/Tasks/TaskBuilder/TaskBuilder';
+import Tasks from './containers/Tasks/Tasks';
+import Login from './containers/Auth/Login/Login';
+import Signup from './containers/Auth/Signup/Signup';
+import Logout from './containers/Auth/Logout/Logout';
+import TaskDetails from './containers/Tasks/TaskDetails/TaskDetails';
+
+class App extends Component {
+  componentDidMount = () => {
+    this.props.onTryAutoSignup();
+  }
+  render() { 
+    return (
+      <div>
+        <Layout>
+          <Switch>
+            <Route path="/todos/create" component={TaskBuilder} />
+            <Route path="/todos/:id" component={TaskDetails} />
```

Comment:
Use dynamic Routing
take navigation meta const
and use React lazy to import dynamic component